Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -294,6 +335,67 @@ public class PBXProjGenerator {
         return targetDependency
     }
 
+    func generateExternalTargetDependency(from: String, to target: String, in project: String) -> (PBXTargetDependency, PBXTarget, PBXReferenceProxy) {
+        guard let projectReference = self.project.getProjectReference(project) else {
+            fatalError("project not found")
+        }
+
+        let pbxProj = try! getPBXProj(from: projectReference)
```

Comment:
I can't import `GenerationError` here because it's in the CLI target. For now I can just make this a throwing function. Or would you prefer a new error type similar to `SchemeGenerationError`?